### PR TITLE
chore: accept database in addition to namespace_name

### PR DESCRIPTION
**Pull request by gavincabbage**
Part of https://github.com/influxdata/idpe/issues/17397

We would like users interacting with e.g. InfluxQL over Flight to be able to use the same `database` param to specify a database that is available when using Flight SQL (added in https://github.com/influxdata/influxdb_iox/pull/7511). Since we also added `bucket` and `bucket-name` to Flight SQL, we also honor those here.

### CODE REVIEW

File: service_grpc_flight/src/request.rs [hunk 1]
```diff
@@ -350,6 +351,24 @@ mod tests {
                 "my_otherdb",
                 "SHOW DATABASES;",
             ),
+            // database metadata
+            TestCase::new_influxql(
```
Comment (alamb) [inline service_grpc_flight/src/request.rs:355]: I think you should also add a new testcase for SQL as well to cover that path as it is quite important

like `TestCase::new_sql(...)`

File: service_grpc_flight/src/request.rs [hunk 2]
```diff
@@ -183,6 +183,7 @@ impl IoxGetRequest {
         /// This represents ths JSON fields
         #[derive(Deserialize, Debug)]
         struct ReadInfoJson {
+            #[serde(alias = "database", alias = "bucket", alias = "bucket-name")]
```
Comment (alamb) [inline service_grpc_flight/src/request.rs:186]: TIL "alias" 👍

**Review by alamb — APPROVED** [review]
The improvement to add support for `database` and `bucket-name` to the JSON looks good to me 👍 

Note there is another encoding (the protobuf encoding) for `ReadInfo`: https://github.com/influxdata/influxdb_iox/blob/41ec673f6fb66977ea7b070d47127d9b788dd084/generated_types/protos/influxdata/iox/querier/v1/flight.proto#L19-L20

I wonder if we want to add the same treatment?

**Comment by gavincabbage** [comment]
> Note there is another encoding (the protobuf encoding) for ReadInfo:

@alamb I went down this road for a bit this morning and the scope of the change quickly expanded beyond my comfort level, so I'll write up an issue for that and we'll schedule proper engineering attention. 

That said, @rickspencer3's examples using InfluxQL over Flight do use the JSON ticket encoding, so this change does have some value on its own, as it lets our examples consistently use "database" and not "namespace". Normally I wouldn't want to merge something half-baked like this, if not for that short-term customer value.

(PS: Sorry for the force push)

**Comment by alamb** [comment]
> That said, @rickspencer3's examples using InfluxQL over Flight do use the JSON ticket encoding, so this change does have some value on its own, as it lets our examples consistently use "database" and not "namespace". Normally I wouldn't want to merge something half-baked like this, if not for that short-term customer value.

I think it makes plenty of sense to merge this PR as is (with just JSON support) and a follow on ticket for protobuf encoding support -- the code in this PR stands as is and the ticket will make sure we don't forget to do the follow on work

🚀

**Pull request merged**